File: arena-backend/src/dtos/message.dto.ts
import { ApiProperty } from "@nestjs/swagger";
import { UserDto } from "./user.dto";

export class MessageDto {
  @ApiProperty({ description: "메시지 ID" })
  messageId: string;

  @ApiProperty({ description: "채널 ID" })
  channelId: string;

  @ApiProperty({ description: "보낸 사람" })
  sender: UserDto;

  @ApiProperty({ description: "메시지 내용" })
  content: string;

  @ApiProperty({ description: "보낸 시간" })
  createdAt: Date;
}

File: apps/backend/src/main/java/app/sandori/arena/api/domain/message/dtos/MessageDto.java
package app.sandori.arena.api.domain.message.dtos;

import app.sandori.arena.api.domain.message.MessageEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDateTime;

@Schema(description = "메시지 응답")
public record MessageDto(
    @Schema(description = "메시지 ID")
    String messageId,

    @Schema(description = "대화방 ID")
    String channelId,

    @Schema(description = "발신자 프로필 ID")
    String senderId,

    @Schema(description = "메시지 내용")
    String content,

    @Schema(description = "생성일시")
    LocalDateTime createdAt,

    @Schema(description = "수정일시")
    LocalDateTime updatedAt
) {
    public static MessageDto from(MessageEntity message) {
        return new MessageDto(
            message.getMessageId(),
            message.getChannelId(),
            message.getSenderId(),
            message.getContent(),
            message.getCreatedAt(),
            message.getUpdatedAt()
        );
    }
}
